picard/ui/matchdetailsdialog/__init__.py

```diff
@@ -0,0 +1,386 @@
+
+    def __init__(self, sort_key):
+        super().__init__()
+        self.sort_key = sort_key
+
+    def __lt__(self, other):
+        return self.sort_key < other.sort_key
+
+
+class MatchDetailsDialogBase(PicardDialog):
+
+    defaultsize = QtCore.QSize(720, 360)
+    autorestore = False
+    scrolled = pyqtSignal()
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.response_results = []
+        self.setupUi()
+        self.restore_state()
+        # self.columns has to be an ordered dict, with column name as keys, and
+        # matching label as values
+        self.columns = None
+        self.sorting_enabled = True
+        self.create_table()
+        self.finished.connect(self.save_state)
+
+    @property
+    def columns(self):
+        return self.__columns
+
+    @columns.setter
+    def columns(self, list_of_tuples):
+        if not list_of_tuples:
+            list_of_tuples = []
+        self.__columns = OrderedDict(list_of_tuples)
+        self.__colkeys = list(self.columns.keys())
+
+    @property
+    def table_headers(self):
+        return list(self.columns.values())
+
+    def colpos(self, colname):
+        return self.__colkeys.index(colname)
+
+    def set_acoustid_lookup_url_edit(self, p_acoustid_lookup_url):
+        self.acoustid_lookup_url.setText("<a href='" + p_acoustid_lookup_url + "'>" + _("Click to see AcoustId Fingerprint Lookup in browser") + "</a> ")
+
+    def set_input_filename_lbl(self, p_filename):
+        self.input_filename_lbl.setText("Input File: " + p_filename)
+
+    def set_table_item(self, row, colname, value, sortkey=None):
+        # QVariant remembers the original type of the data
+        # matching comparison operator will be used when sorting
+        # get() will return a string, force conversion if asked to
+        if sortkey is None:
+            sortkey = natsort.natkey(value)
+        item = SortableTableWidgetItem(sortkey)
+        item.setData(QtCore.Qt.DisplayRole, value)
+        pos = self.colpos(colname)
+        if pos == 0:
+            item.setData(QtCore.Qt.UserRole, value)
+        self.table.setItem(row, pos, item)
+
+    def setupUi(self):
+        self.verticalLayout = QtWidgets.QVBoxLayout(self)
+        self.verticalLayout.setObjectName("vertical_layout")
+
+        self.input_filename_widget = QtWidgets.QWidget(self)
+        self.input_filename_lbl = QtWidgets.QLabel(self.input_filename_widget)
+        self.input_filename_lbl.setFocusPolicy(QtCore.Qt.StrongFocus)
+        myFont = QtGui.QFont()
+        myFont.setBold(True)
+        self.input_filename_lbl.setFont(myFont)
+        self.input_filename_lbl.setTextInteractionFlags(QtCore.Qt.TextSelectableByMouse)
+        self.verticalLayout.addWidget(self.input_filename_lbl)
+
+        self.tip_widget = QtWidgets.QWidget(self)
+        self.tip_lbl = QtWidgets.QLabel(self.tip_widget)
+        self.tip_lbl.setText(_("\nTip: Column header left click to sort, right click to filter"))
```

Comment:
Is this `\n` strictly needed?

Reply:
No, I think its left over from some of the json info I was previously showing.  I removed it.